### ISSUE TITLE
`#partial_update_objects` fails to accept options

### ISSUE THREAD

**Issue by vincedevendra** [algolia/algoliasearch-client-ruby#470]
- Algolia Client Version: 2.2.2
- Language Version: Ruby 2.7.5

### Description

I'm getting 400 errors when trying to call `index.partial_update_objects`. See repro steps for more details.

### Steps To Reproduce
Run the following:
```ruby
client = Algolia::Search::Client.create(<app_id>, <api_key>)
index = client.init_index(<index_name>)

index.partial_update_objects([objectID: '123', attribute: 'value'], {createIfNotExists: true})
  # raises Algolia::AlgoliaHttpError (400: Invalid object attributes:createIfNotExists near line:1 column:114)
```

**Comment by OlivierLeCloset** [comment]
Hi,
Same error for me. It seems to be linked to this PR https://github.com/algolia/algoliasearch-client-ruby/pull/458.
Some news ?

**Comment by nooysters** [comment]
What are the chances of PR https://github.com/algolia/algoliasearch-client-ruby/pull/458 getting merged?
We are still on V1 but want to upgrade to V2 but we use the partial updates and need createIfNotExists to work.

**Comment by DevinCodes** [comment]
Hi all!

I think #458 may fix this indeed. We've released this in v2.3.0. I'm closing this issue for now, feel free to reopen in case the issue persists after upgrading.

Cheers!
